src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt

```diff
@@ -73,21 +71,15 @@ data class RustToolchain(val location: String) {
         if (procOut.exitCode != 0 || procOut.isCancelled || procOut.isTimeout)
             return null
 
-        return extractor(procOut.stdoutLines)
+        return procOut.stdoutLines
     }
 
-    fun cargo(cargoProjectDirectory: String): Cargo =
-        Cargo(pathToExecutable(CARGO), pathToExecutable(RUSTC), cargoProjectDirectory)
-
-    fun nonProjectCargo(): Cargo =
-        Cargo(pathToExecutable(CARGO), pathToExecutable(RUSTC), null)
-
-    val presentableLocation: String = PathUtil.toPresentableUrl(pathToExecutable(CARGO))
-
-    private fun pathToExecutable(toolName: String): String {
-        val exeName = if (SystemInfo.isWindows) "$toolName.exe" else toolName
-        return File(File(location), exeName).absolutePath
-    }
+    data class VersionInfo(
```

Comment:
hmm, creating UI control in logic and passing it all around doesn't seem like a good idea imho